src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1429,7 +1429,7 @@ void TorrentImpl::forceRecheck()
     {
         // When "force recheck" is applied on paused torrent, we temporarily resume it
         resume();
-        setStopCondition(StopCondition::FilesChecked);
```

Comment:
`setStopCondition()` performs some precondition checks including for `isChecking()` result which is assumed to be `true` but is still `false` in this point because libtorrent starts checking asynchronously and only then the status is updated to reflect it. But here we are enough that `force_recheck()` was already invoked, thus we can avoid any precondition checks.